boot/uboot-mediatek: Fix ramips/mt76x8 buildbot

jdeisenh opened the pull request:
The build hangs in make syncconfig for u-boot-ravpower_rp-wd009 
Because CONFIG_SYS_MIPS_TIMER_FREQ is not in the .config, the build hangs in a console prompt for the value. 
Move defines from header to defconfig

Signed-off-by: Jo Deisenhofer <jo.deisenhofer@gmail.com>

jdeisenh committed:
boot/uboot-mediatek: Fix ramips/mt76x8 buildbot

Fixes: 3d5c5427e17a ("uboot-mediatek: update to U-Boot 2023.01")

Move defines from header to defconfig
The package build and the Buildbot hang in 'make syncconfig' for u-boot-ravpower_rp-wd009 because CONFIG_SYS_MIPS_TIMER_FREQ is not in the .config, causing a console prompt.
Also moved two other defines in defconfig causing duplicate definition
warnings.

Signed-off-by: Jo Deisenhofer <jo.deisenhofer@gmail.com>

Ansuel commented on package/boot/uboot-mediatek/patches/420-add-support-for-RAVPower-RP-WD009.patch:
```diff
@@ -140,6 +140,7 @@ Subject: [PATCH] add support for RAVPower RP-WD009
 +CONFIG_SYS_CONSOLE_INFO_QUIET=y
 +CONFIG_VERSION_VARIABLE=y
 +CONFIG_BOARD_RAVPOWER_RP_WD009=y
++CONFIG_SYS_MIPS_TIMER_FREQ=290000000
```
not documented change?

jdeisenh replied:
Because this value is not missing the .config, the make config run hangs and prompts for the value. 
This prompt is about 1Mio times repeated in the log output of the runner, until it timeouts. 
`Build U-Boot image with exception vectors (ROM_EXCEPTION_VECTORS) [Y/?] y
Fixed MIPS CPU timer frequency in Hz (SYS_MIPS_TIMER_FREQ) [] (NEW) 
`
The value comes from the include file of the same target. 
I'll expand the documentation

Ansuel commented:
@jdeisenh do you have some additional info about this problem and why we need this and how you came with this solution?

Also commit title should not have dot at the end
And a new line is required after the end of the commit description and the start of the sob tag

Also does this fix any reported issue or commit?

xabolcs commented:
@jdeisenh the hang is caused by the latest uboot-mediatek version bump in commit 3d5c5427e17a870041e0c94b6fbbfb9301cd2c70?

jdeisenh commented:
> @jdeisenh do you have some additional info about this problem and why we need this and how you came with this solution?
> 
The problem is that buildbot for ramips/mt76x8 hangs since a few days with a timeout. 
The problem can be seen when building a mt76x8 target, doing
make package/boot/uboot-mediatek/clean
make package/boot/uboot-mediatek/compile -j 1 V=cs

It will hang at with a prompt

* General setup
*
Build U-Boot image with exception vectors (ROM_EXCEPTION_VECTORS) [Y/?] y
Fixed MIPS CPU timer frequency in Hz (SYS_MIPS_TIMER_FREQ) [] (NEW) 

This line is repeated in the build logs a million times, until it timeouts

> Also commit title should not have dot at the end And a new line is required after the end of the commit description and the start of the sob tag
>

Fixed.
 
> Also does this fix any reported issue or commit?

I am not aware. I have add a new target, and wondered why it did not show up in snapshots

jdeisenh commented:
> @jdeisenh the hang is caused by the latest uboot-mediatek version bump (see commit [3d5c542](https://github.com/openwrt/openwrt/commit/3d5c5427e17a870041e0c94b6fbbfb9301cd2c70))?

Likely since the timing matches. I havent bisected it.

xabolcs commented:
> I havent bisected it.

Checking out latest `master` (currently afcf1a4de4), reverting that 3d5c5427e17a870041e0c94b6fbbfb9301cd2c70 commit, retrying building uboot-mediatek on `mt76x8` target: SUCCESS.

So thank you @jdeisenh for testing on MIPS target, the original commit noted it wasn't tested there.

xabolcs commented:
> > Also commit title should not have dot at the end And a new line is required after the end of the commit description and the start of the sob tag
> 
> Fixed.

You just updated the PR's description, but not the commit! :slightly_smiling_face: 
Please (force) push the updated commit too! And being there, please add that `Fixes:` tag too!

dangowrt commented:
Pushed to master, thank you for taking care of this before I even noticed...
